### ISSUE TITLE
Arangosearch: PHRASE function combined with other filter not working

### ISSUE THREAD

**Issue by Darkheir** [arangodb/arangodb#10090]
## My Environment

* __ArangoDB Version__:        3.5.0
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:   anual Start in Docker
* __Infrastructure__:               local
* __Total RAM in your machine__:        16Gb
* __Disks in use__:        SSD
* __Used Package__:              Docker - official Docker library

## Component, Query & Data

__Affected feature__: ArangoSearch

__Search view__:

```json
{
  "links": {
    "my_collection": {
      "analyzers": [
        "identity"
      ],
      "fields": {
        "description": {
          "analyzers": [
            "text_en",
            "identity"
          ]
        },
        "type": {},
        "name": {
          "analyzers": [
            "text_en",
            "identity"
          ]
        }
      },
      "includeAllFields": false,
      "storeValues": "none",
      "trackListPositions": false
    }
  }
}
```

__AQL query (if applicable)__:

```
FOR item IN my_view
    SEARCH PHRASE(item.name, "foo bar", 'text_en') AND item.type == "my_type"
    RETURN item
```

__AQL explain (if applicable)__:

```
Execution plan:
 Id   NodeType            Est.   Comment
  1   SingletonNode          1   * ROOT
  2   EnumerateViewNode      1     - FOR item IN my_view SEARCH ((item.`type` == "my_type") && PHRASE(item.`name`, "foo bar", "text_en"))   /* view query */
  3   ReturnNode             1       - RETURN item

Indexes used:
 none

Optimization rules applied:
 Id   RuleName
  1   handle-arangosearch-views
```

## Steps to reproduce

The query above returns no results and should return one.

The following queries all return the expected result:

__Only one word in PHRASE search__:

```
FOR item IN my_view
    SEARCH PHRASE(item.name, "foo", 'text_en') AND item.type == "my_type"
    RETURN item
```

__No filter on type__:

```
FOR item IN my_view
    SEARCH PHRASE(item.name, "foo bar", 'text_en')
    RETURN item
```

__Using TOKEN instead of PHRASE__:

```
FOR item IN objects_search_view
    SEARCH ANALYZER(item.name in TOKENS("foo bar", 'text_en'), "text_en") AND item.type == "my_type"
    RETURN item
```

**Comment by gnusi** [comment]
Hi @darkheir,

Thank you for submitting the issue. Could you please share the dataset you use so we can reproduce the problem on our side?

**Comment by Darkheir** [comment]
I managed to reproduce the issue with this small set:

```json
[ 
   { 
      "_key":"identity--357447d7-9229-4ce1-b7fa-f1b83587048e",
      "_id":"my_collection/identity--357447d7-9229-4ce1-b7fa-f1b83587048e",
      "_rev":"_ZUy9LSS---",
      "name":"SEKOIA",
      "type":"identity"
   },
   { 
      "_key":"identity--1621d4d4-b67d-41e3-9670-f01faf20d111",
      "_id":"my_collection/identity--1621d4d4-b67d-41e3-9670-f01faf20d111",
      "_rev":"_ZUy9LSS--A",
      "name":"Adversary Bravo",
      "type":"identity"
   }
]
```

I notice that the `_rev` fields look very similar. Could it be the issue ?

**Comment by aveiros** [comment]
If you insert into the collection using this order, you get 0 results
doc1 => { "type": "person", "text": "foo" }
doc2 => { "type": "person", "text": "foo bar" }

Reversing insertion results, you get 1 result.

**Comment by Darkheir** [comment]
I agree with you @aveiros If I insert the documents in the opposite order then the search is working fine.

**Comment by gnusi** [comment]
@Darkheir @aveiros thank you for your effort to reproducing this, we'll address the issue soon. I will keep you updated.

**Comment by iurii-i-popov** [comment]
The fix will be available in the next minor release.

**Comment by aveiros** [comment]
tested against docker 3.5 preview build and looks good :)

**Comment by maxkernbach** [comment]
Hi @Darkheir and @aveiros,

Version 3.5.3, which includes a fix for this issue, has been released.

From the [Changelog](https://github.com/arangodb/arangodb/blob/660ff994dbfa994a18fe921481f591b878b51619/CHANGELOG#L133):

> * Fixed issue #10090: fix repeatable seek to the same document in
  SEARCH operations for ArangoSearch views.

Please find the latest release in the [download section](https://www.arangodb.com/download-major/).

**Comment by Darkheir** [comment]
Thank you for the release! We installed it and it fixes our issue :-)
